sdk/core/System.ClientModel/src/Pipeline/ClientRetryPolicy.cs

```diff
@@ -0,0 +1,238 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System.ClientModel.Internal;
+using System.Collections.Generic;
+using System.IO;
+using System.Runtime.ExceptionServices;
+using System.Threading;
+using System.Threading.Tasks;
+
+namespace System.ClientModel.Primitives;
+
+public class ClientRetryPolicy : PipelinePolicy
+{
+    private const int DefaultMaxRetries = 3;
+    private static readonly TimeSpan DefaultInitialDelay = TimeSpan.FromSeconds(0.8);
+
+    private readonly int _maxRetries;
+    private readonly TimeSpan _initialDelay;
+
+    public ClientRetryPolicy() : this(DefaultMaxRetries)
+    {
+    }
+
+    public ClientRetryPolicy(int maxRetries)
+    {
+        _maxRetries = maxRetries;
+        _initialDelay = DefaultInitialDelay;
+    }
+
+    public sealed override void Process(PipelineMessage message, IReadOnlyList<PipelinePolicy> pipeline, int currentIndex)
+        => ProcessSyncOrAsync(message, pipeline, currentIndex, async: false).EnsureCompleted();
+
+    public sealed override async ValueTask ProcessAsync(PipelineMessage message, IReadOnlyList<PipelinePolicy> pipeline, int currentIndex)
+        => await ProcessSyncOrAsync(message, pipeline, currentIndex, async: true).ConfigureAwait(false);
+
+    private async ValueTask ProcessSyncOrAsync(PipelineMessage message, IReadOnlyList<PipelinePolicy> pipeline, int currentIndex, bool async)
+    {
+        List<Exception>? allTryExceptions = null;
```

Comment:
Does adding `Try` to this and the other variables add more context?

Reply:
I think when I originally did this work, it took me a little while to get my head around the difference between "exceptions" and "lastException" given the way this method is structured and how the references to both of them are interleaved.  I found `allTryExceptions` and `thisTryException` a little more descriptive and helpful as I was working to integrate this functionality with Azure.Core -- I considered adding a comment but that wasn't helping me track the plot further down in the file.  That said, I don't feel super strongly about a single variable name, so if you care a lot about this one, I'm happy to rename it to something you like better -- feel free to suggest an alternative!